### ISSUE TITLE
New class and feature?

### ISSUE THREAD

**Issue by BugsOverBugs** [ivofernandes/yahoo_finance_data_reader#2]
Hey, I have created a data class for your package (See below). 
It works like this:

```dart
List<YahooFinDateData> data = (snapshot?.data ?? []).map((e) => YahooFinDateData.fromJson(e)).toList();
```

- [x] #3
- [x] #4

It could look like this:
```dart
YahooFinanceDailyReader().getDataFromDate(ticker, DateTime.now());
```

Thanks!

```dart
class YahooFinDateData {
  final dynamic json;
  final DateTime date;
  final double? open;
  final double? high;
  final double? low;
  final double? close;
  final int? volume;
  final double? adjClose;

  YahooFinDateData(
    this.json, {
    required this.date,
    required this.open,
    required this.high,
    required this.low,
    required this.close,
    required this.volume,
    required this.adjClose,
  });

  static YahooFinDateData fromJson(json) {
    return YahooFinDateData(
      json,
      date: DateTime.fromMillisecondsSinceEpoch(json['date'] * 1000),
      open: json['open'] == null ? null : double.parse(json['open'].toString()),
      high: json['high'] == null ? null : double.parse(json['high'].toString()),
      low: json['low'] == null ? null : double.parse(json['low'].toString()),
      close:
          json['close'] == null ? null : double.parse(json['close'].toString()),
      volume:
          json['volume'] == null ? null : int.parse(json['volume'].toString()),
      adjClose: json['adjclose'] == null
          ? null
          : double.parse(json['adjclose'].toString()),
    );
  }

  @override
  String toString() => json;
}
```

**Comment by ivofernandes** [comment]
Amazing :) thanks for the feedback and for the code!! Will add that functionality.

I would see it as an append to the existing functionality. Because if we get 20000 rows for all the ^gpsc what we really don't want is that we are forced to do some computation besides the already needed json.parse

But I will add it as an extra method

**Comment by ivofernandes** [comment]
About the 

YahooFinanceDailyReader().getDataFromDate(ticker, DateTime.now());

there is already a parameter for start time
  Future<List<dynamic>> getDailyData(String ticker,
      {int startTimestamp = -1577908800}) async {
      
So is just a matter of passing DateTime.now().millisecondsSinceEpoch / 1000, but yeah the developer experience may be improved if we can pass a startDateTime instead of the startTimestamp

**Comment by ivofernandes** [comment]
Already added the date and tomorrow I will submit a new version of the package, will keep the existing interface, and add a new one, this way I can follow the open closed principle and don't break existing projects that depend on this package, then in the next weeks will try to adjust the package example to have all the functionality :) 

https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle

Thanks again for your super useful feedback :)

**Comment by ivofernandes** [comment]
Just released the new DTO and DateTime in version 0.0.7

https://pub.dev/packages/yahoo_finance_data_reader
